lib/sidekiq_unique_jobs/lock/until_and_while_executing.rb

```diff
@@ -16,9 +16,8 @@ class UntilAndWhileExecuting < BaseLock
       # Executes in the Sidekiq server process
       # @yield to the worker class perform method
       def execute
-        return unless locked?
+        unlock if locked?
```

Comment:
it should really be `return unless unlock`. The problem with this PR and the change in logic is that the will run even if another process (job_id) has the lock rendering the first part completely useless.

If you have a look in the locksmith class `unlock` method it actually checks if it is `locked and allowed to unlock. See my comment in the test

Reply:
Thanks for the quick feedback! I understand your argument. Since the `WhileExecuting` server lock is still there I thought this was a minor problem since we will still be preventing simultaneous execution.

Excuse my ignorance, but how would you properly check if this was locked by another job?

Reply:
There are a couple of changes I have in mind. I am currently working on improving the reliability of the gem, more specifically I am trying to avoid keys being left to rot in Redis.

I'll take your suggestion and improve on it in #387.

Reply:
Regarding why put it in the LUA script:

Since we are talking about high concurrency and LUA is transactional we can only really verify the ownership of the lock there.

Reply:
Perfect, thanks. Let's close this guy then in favor of #387